src/pip/_internal/wheel.py

```diff
@@ -773,52 +773,84 @@ def _contains_egg_info(
     return bool(_egg_info_re.search(s))
 
 
-def should_use_ephemeral_cache(
+def should_build(
     req,  # type: InstallRequirement
+    need_wheel,  # type: bool
     format_control,  # type: FormatControl
-    should_unpack,  # type: bool
-    cache_available  # type: bool
 ):
-    # type: (...) -> Optional[bool]
     """
-    Return whether to build an InstallRequirement object using the
-    ephemeral cache.
-
-    :param cache_available: whether a cache directory is available for the
-        should_unpack=True case.
-
-    :return: True or False to build the requirement with ephem_cache=True
-        or False, respectively; or None not to build the requirement.
+    Return whether a wheel should be built based on the requirement
+    characteristics, and if a wheel is ultimately needed or not.
     """
     if req.constraint:
         # never build requirements that are merely constraints
-        return None
+        return False
+
     if req.is_wheel:
-        if not should_unpack:
+        if need_wheel:
             logger.info(
                 'Skipping %s, due to already being wheel.', req.name,
             )
-        return None
-    if not should_unpack:
-        # i.e. pip wheel, not pip install;
-        # return False, knowing that the caller will never cache
-        # in this case anyway, so this return merely means "build it".
-        # TODO improve this behavior
+        # never build a requirement that is already a wheel
         return False
 
-    if req.editable or not req.source_dir:
-        return None
+    if need_wheel:
+        return True
+
+    if req.editable:
+        # don't build an editable if no wheel is needed
+        return False
+
+    if not req.source_dir:
+        # TODO explain what no source_dir means
+        return False
 
     if "binary" not in format_control.get_allowed_formats(
             canonicalize_name(req.name)):
         logger.info(
             "Skipping bdist_wheel for %s, due to binaries "
             "being disabled for it.", req.name,
         )
-        return None
+        return False
+
+    return True
+
+
+def should_use_ephemeral_cache(
```

Comment:
Done. I also removed the `cache_available` parameter from `should_cache` because it does not make sense to ask if we can cache if we have no cache available. So it's clearer to test for `cache_available` at the call site.